lib/init.php

```diff
@@ -57,3 +57,121 @@ function gutenberg_menu() {
 	);
 }
 add_action( 'admin_menu', 'gutenberg_menu', 9 );
+
+if ( ! function_exists( 'html_contains_block' ) ) {
+	/**
+	 * Returns whether the given HTML contains a block
+	 * of the given type and, if provided,
+	 * a given attribute and attribute value.
+	 * 
+	 * Note that it's not possible to search for an attribute
+	 * whose value is `null`.
+	 *
+	 * @param  string  $html The html to search in.
+	 * @param  string  $block_name      Find this block type,
+	 *                                  with an optional "core/" namespace,
+	 *                                  e.g. "paragraph", "core/paragraph",
+	 *                                  "my_plugin/my_block".
+	 * @param  string  $attribute_name  If provided, the block must also
+	 *                                  contain this attribute.
+	 * @param  string  $attribute_value If provided, the given attribute's
```

Comment:
nitpick: we'll want to add `?string` (I'm pretty sure vs. `string|null`) before merging